src/main.rs

```diff
@@ -22,6 +25,30 @@ pub struct GordoSpec {
     config: Value,
 }
 
+#[derive(Deserialize, Debug)]
+pub struct GordoControllerConfigMap {
+    deploy_image: String,
+}
+
+impl Default for GordoControllerConfigMap {
+    fn default() -> Self {
+        GordoControllerConfigMap {
+            deploy_image: "auroradevacr.azurecr.io/gordo-infrastructure/gordo-deploy".to_owned(),
+        }
+    }
+}
+impl From<BTreeMap<String, String>> for GordoControllerConfigMap {
+    fn from(data: BTreeMap<String, String>) -> Self {
+        let default_config = GordoControllerConfigMap::default();
```

Comment:
I think the usual way is to let the deployment mount in the values from the configmap as enviroment variables as in this example: https://blog.bigbinary.com/2017/05/25/using-kubernetes-configmap-with-configuration-files-for-deploying-rails-app.html  Then 
1. The program dont need to know about the configmap at all, and 
2. It is visible from `kubectl describe deployment ...` what configuration is used.